keda/templates/cert-manager/self-ca.yaml

```
@@ -13,8 +13,7 @@ spec:
   privateKey:
     algorithm: RSA
     size: 2048
-  duration: 8760h0m0s # 1 year
-  renewBefore: 720h0m0s # 1 month
+  duration: 43800h0m0s # 5 years
```

with the leaf cert having back `renewBefore` which means they rotate every 4 months, the CA should rotate itself at least 5 months before it expires. Given it's now going to be signed for 5 years, to be conservative we can set `renewBefore` for CA to 1 year, meaning it will rotate every 4 years.

calculated the hours for leap year 366*24
```suggestion
  duration: 43800h0m0s    # 5 years
  renewBefore: 8784h0m0s  # 1 year
```

---

Is this cutting it too short still?

All it takes is for one user to bump the duration of the leaf more than 2 months and we are back at the same problem.

Removing a config is a win win in my opinion but if you are fixed on it I am happy to change it

---

> Is this cutting it too short still?

maybe my math is off, let me elaborate on my logic here:
* the cert is signed for a year and is renewed 8 months before expiry => rotates every 4 months
* as long as the CA will continue to be valid for at least 4 months when signing the cert, the bug should not appear
* this suggests that a year before CA's expiration, it should rotate, neaning when it signs the last cert, it will still have 1 year validity
* the certs during that year will rotate twice already and get signed by the new CA

does this sound right? or am I missing a few months somewhere again...

> All it takes is for one user to bump the duration of the leaf more than 2 months and we are back at the same problem.

unfortunately, with the options given by the chart, we can't easily prevent users from shooting themself in the foot, but at least we can make it not broken by default

> Removing a config is a win win in my opinion

from a purely technical perspective, perhaps it would be better, but opensource projects have to worry about maintaining backwards compatibilty. Removing a configuration option from a chart will have to go through [a deprecation period](https://github.com/kedacore/governance/blob/3c60f829e4660e40ace301ce316ce3a6f1783d83/DEPRECATIONS.md?plain=1#L52). That means it can still be done, but will take a few releases. Meanwhile, we can fix the defaults so no other user gets this unpleasant surprise.

---

I just got what you meant here, keeping the leaf cert configurable but the CA instead of rotating 1 year before expiration, rotating default 2/3 of duration. I'm ok with this :)

---

hmm, one more thing, let's make the `renewBefore` explicit, we can make it `14600h0m0s` which should be 1/3 of 5 years. If someone used this chart with argo or flux and enabled server-side apply, the `720h0m0s` would remain there I think.

---

Sorry i couldnt response i was busy on a work trip.

LGTM. Thanks for merging 😁